### ISSUE TITLE
has hlint relaxed ?

### ISSUE THREAD

**Issue by juhp** [ndmitchell/hlint#1189]
Is it me or has hlint become more relaxed in 2020?
I feel like I am getting far fewer hlint suggestions than earlier.

Just one example I noticed today: hlint didn't flag this code for example:

```
whenJust mdir $ \ dir -> setCurrentDirectory dir
```
which I think it would have done in the past?  Admittedly I haven't tried older versions yet to verify.
If it is trying to stress us less during 2020 that's okay 😂

**Comment by ndmitchell** [comment]
HLint has relaxed a bunch. I still expected `\x -> foo x` to be reduced, but there are various type checker changes in GHC which mean that eta reduction keeps becoming gentler. I'm wondering if this one should still be hinted about though.

**Comment by juhp** [comment]
I don't mind... 

Another one that bothers me is redundant `do` usage.

**Comment by zliu41** [comment]
> whenJust mdir $ \ dir -> setCurrentDirectory dir

I think this has been case [for about 15 years](https://github.com/ndmitchell/hlint/commit/dc98403c3f7b448271c3222abc953c954c1b3ec5) :smirk:

I suggest we allow removing the lambda, if (1) the new expression is not an `OpApp`, or (2) the parent operator is `$`.
